lib/ansible/inventory/manager.py

```diff
@@ -161,13 +165,59 @@ def __init__(self, loader, sources=None, parse=True, cache=True):
         else:
             self._sources = sources
 
+        self._launch_ssh_agent()
+
         # get to work!
         if parse:
             self.parse_sources(cache=cache)
 
         self._cached_dynamic_hosts = []
         self._cached_dynamic_grouping = []
 
+    def _launch_ssh_agent(self):
+        ssh_agent_cfg = C.config.get_config_value('SSH_AGENT')
+        match ssh_agent_cfg:
+            case 'none':
+                return
+            case 'auto':
+                ssh_agent_dir = os.path.join(C.DEFAULT_LOCAL_TMP, 'ssh_agent')
+                os.mkdir(ssh_agent_dir, 0o700)
+                try:
+                    ssh_agent_bin = get_bin_path('ssh-agent', required=True)
+                except ValueError as e:
+                    raise AnsibleError('SSH_AGENT set to auto, but cannot find ssh-agent binary') from e
+                sock = os.path.join(ssh_agent_dir, 'agent.sock')
+                p = subprocess.Popen(
+                    [ssh_agent_bin, '-D', '-s', '-a', sock],
+                    stdin=subprocess.PIPE,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                )
+                if p.poll() is not None:
+                    raise AnsibleError(
+                        f'Could not start ssh-agent: (rc={p.returncode}) {p.stderr}'
+                    )
+                if (stdout := p.stdout.read(13)) != b'SSH_AUTH_SOCK':
+                    display.warn(
+                        f'The first 13 characters of stdout did not match the '
+                        f'expected SSH_AUTH_SOCK. This may not be the right binary, '
+                        f'or an incompatible agent: {stdout.decode()}'
+                    )
+                display.vvv(f'SSH_AGENT: ssh-agent[{p.pid}] started and bound to {sock}')
+                atexit.register(p.terminate)
+            case _:
+                sock = ssh_agent_cfg
+
+        try:
+            with SshAgentClient(sock) as client:
+                client.list()
+        except Exception as e:
+            raise AnsibleError(
+                f'Could not communicate with ssh-agent using auth sock {sock}: {e}'
+            )
```

Comment:
Do we know which exceptions are possible here? Also, best to use an explicit `from`.
```suggestion
        except Exception as arbitrary_err:
            raise AnsibleError(
                f'Could not communicate with ssh-agent using auth sock {sock}: {arbitrary_err}'
            ) from arbitrary_err
```